src/Tickets/Commerce/Repositories/Order_Repository.php

```diff
@@ -487,4 +487,254 @@ public function filter_by_events_not( $events = null ) {
+	}
+
+	/**
+	 * Sets up the query filters to order items by the gateway id.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $order      The order direction, either `ASC` or `DESC`; defaults to `null` to use the order
+	 *                           specified in the current query or default arguments.
+	 * @param bool   $after      Whether to append the duration ORDER BY clause to the existing clauses or not;
+	 *                           defaults to `false` to prepend the duration clause to the existing ORDER BY
+	 *                           clauses.
+	 * @param bool   $override   Whether to override existing ORDER BY clauses with this one or not; default to
+	 *                           `true` to override existing ORDER BY clauses.
+	 */
+	protected function order_by_gateway_id( $order = null, $after = false, $override = true ) {
+		$this->order_by_key( Order::$gateway_order_id_meta_key, $order, $after, $override );
+	}
+
+	/**
+	 * Sets up the query filters to order items by the gatway name.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $order      The order direction, either `ASC` or `DESC`; defaults to `null` to use the order
+	 *                           specified in the current query or default arguments.
+	 * @param bool   $after      Whether to append the duration ORDER BY clause to the existing clauses or not;
+	 *                           defaults to `false` to prepend the duration clause to the existing ORDER BY
+	 *                           clauses.
+	 * @param bool   $override   Whether to override existing ORDER BY clauses with this one or not; default to
+	 *                           `true` to override existing ORDER BY clauses.
+	 */
+	protected function order_by_gateway( $order = null, $after = false, $override = true ) {
+		$this->order_by_key( Order::$gateway_meta_key, $order, $after, $override );
+	}
+
+	/**
+	 * Sets up the query filters to order items by the status.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $order      The order direction, either `ASC` or `DESC`; defaults to `null` to use the order
+	 *                           specified in the current query or default arguments.
+	 * @param bool   $after      Whether to append the duration ORDER BY clause to the existing clauses or not;
+	 *                           defaults to `false` to prepend the duration clause to the existing ORDER BY
+	 *                           clauses.
+	 * @param bool   $override   Whether to override existing ORDER BY clauses with this one or not; default to
+	 *                           `true` to override existing ORDER BY clauses.
+	 */
+	protected function order_by_status( $order = null, $after = false, $override = true ) {
+		global $wpdb;
+
+		$meta_alias = 'status';
+		$filter_id = "order_by_{$meta_alias}";
+		$prefix = 'tec-' . Commerce::ABBR . '-';
+
+		$order = $order === null
+			? Arr::get_in_any( [ $this->query_args, $this->default_args ], 'order', 'ASC' )
```

Comment:
I see that we're using this one on the other PR as well many times. Would it be worth it to create a small method?